### PR TITLE
[docs] clean up documentation based on latest clusterctl release

### PR DESCRIPTION
**Pull request by eljohnson92**
**What this PR does / why we need it**:
This PR cleans up the clusterctl config references now that there is a released version of clusterctl that includes the `linode-linode` provider
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [x] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

**Review by amold1 — DISMISSED** [review]

**Review by amold1 — DISMISSED** [review]

File: e2e/gha-clusterctl-config.yaml [hunk 1]
```diff
@@ -1,16 +1,10 @@
...
     type: InfrastructureProvider
   - name: "k3s"
     url: https://github.com/k3s-io/cluster-api-k3s/releases/latest/bootstrap-components.yaml
     type: "BootstrapProvider"
   - name: "k3s"
     url: https://github.com/k3s-io/cluster-api-k3s/releases/latest/control-plane-components.yaml
     type: "ControlPlaneProvider"
-  - name: "rke2"
-    url: "https://github.com/rancher-sandbox/cluster-api-provider-rke2/releases/latest/bootstrap-components.yaml"
-    type: "BootstrapProvider"
```
Comment (rahulait) [inline e2e/gha-clusterctl-config.yaml:13]: Does this mean we don't want to perform tests for rke2?

Comment (eljohnson92) [reply e2e/gha-clusterctl-config.yaml:13]: This config is built into recent versions of clusterctl so doesn't need to be defined here

**Comment by rahulait** [comment]
I don't see ccm-linode coming up on rke2 and k3s after these changes. rke2 and k3s has nodeSelector as `node-role.kubernetes.io/control-plane=true` and we are adding `node-role.kubernetes.io/control-plane=`

File: templates/flavors/k3s/default/k3sControlPlane.yaml [hunk 2]
```diff
@@ -15,29 +15,6 @@ spec:
...
-            bootstrap: true
-            valuesContent: |-
-              routeController:
-                vpcName: ${VPC_NAME:=${CLUSTER_NAME}}
-                clusterCIDR: 10.0.0.0/8
-                configureCloudRoutes: true
-              secretRef:
-                name: "linode-token-region"
-              nodeSelector:
-                node-role.kubernetes.io/control-plane: "true"
```
Comment (rahulait) [inline templates/flavors/k3s/default/k3sControlPlane.yaml:40]: Looks like this node-selector is not getting set correctly when using templates/addons

Comment (rahulait) [reply templates/flavors/k3s/default/k3sControlPlane.yaml:40]: Might need a patch to set the nodeSelector to correct value. Default is set to blank. https://github.com/linode/linode-cloud-controller-manager/blob/main/deploy/chart/values.yaml#L16

**Review by rahulait — APPROVED** [review]
LGTM

**Review by amold1 — APPROVED** [review]

**Pull request merged**